lib/bounce_email.rb

```diff
@@ -237,5 +239,38 @@ def extract_original_message_after_delimiter(mail, delimiter_index)
       message = mail.body.to_s.split(delimiter).last
       message.split(INLINE_MESSAGE_END_DELIMITER).first.strip if message.match(INLINE_MESSAGE_END_DELIMITER)
     end
+
+    def extract_and_assign_fields_from_original_mail(mail)
+      extract_and_assign_message_id(mail) if mail.message_id.nil?
+      extract_and_assign_from(mail) if mail.from.nil?
+      extract_and_assign_to(mail) if mail.to.nil?
+      extract_and_assign_subject(mail) if mail.subject.nil?
+      mail
+    end
+
+    def extract_and_assign_message_id(mail)
```

Comment:
Ah, haha, I didn't think of that :)

To be honest, I'm not sure if we need all these helper methods. What do you think if we go this way:

```ruby
def extract_and_assign_fields_from_original_mail(mail)
  if mail.message_id.nil?
    mail.add_message_id extract_field_from(mail, 'Message-ID:')
  end

  if mail.from.nil?
    mail.from = extract_field_from(mail, 'From:')
  end

  if mail.to.nil?
    mail.to = extract_field_from(mail, 'To:')
  end

  if mail.subject.nil?
    mail.subject = extract_field_from(mail, 'Subject:')
  end

  mail
end
```